### ISSUE TITLE
Very slow process

### ISSUE THREAD

**Issue by ntromas** [cmks/DAS_Tool#88]
Hi!
First of all thanks for the pipeline!
I am trying to run DAS_tool with few MAGs from 4 samples, 2 binners (Vamb and Metabat2) and for unknown reason the pipeline got stuck at some point.

Here is the call to DASTool we've been using, running in a singularity container. Early on it works well and it makes good use of all the threads I give it. Annotating the genes and such saturates my CPU, then when it gets to dereplicating the bins and etc it goes more slowly, only appearing to use a couple of threads (not nearly saturating the amount of CPU threads I've given it).

        DASTool -i /wd/vamb-bins/vamb-bins.tsv,/wd/metabat-bins/metabatbins.tsv \
        -c /wd/assembly/co-assembly.fasta \
        -l vamb,metabat2 \
        -o das_bin \
        --search_engine=blastp \
        --write_bin_evals \
        --write_bins \
        --score_threshold=0.65 \
        --threads=16

Output:

bin    bin_set    unique_SCGs    redundant_SCGs    SCG_set    size    contigs    N50    bin_score    SCG_completeness    SCG_redundancy
metabat.8    metabat    51    0    bacteria    2110148    24    144036    1    100    0
tetraESOM.83    tetraESOM    51    0    bacteria    2712894    44    90886    1    100    0
metabat.11    metabat    51    0    bacteria    3295025    70    69995    1    100    0
maxbin2.032    maxbin    51    0    bacteria    2323359    107    33706    1    100    0
tetraESOM.5    tetraESOM    50    0    bacteria    1912208    33    104909    0.980392156862745    98    0
concoct.17    concoct    50    0    bacteria    2259436    57    65532    0.980392156862745    98    0
concoct.77    concoct    50    0    bacteria    1909225    67    47497    0.980392156862745    98    0
tetraESOM.26    tetraESOM    49    0    bacteria    1930369    26    125506    0.96078431372549    96    0
tetraESOM.17    tetraESOM    49    0    bacteria    2117566    36    102700    0.96078431372549    96    0
maxbin2.017    maxbin    41    0    bacteria    2363932    47    155495    0.803921568627451    80    0
concoct.60    concoct    42    1    bacteria    1814275    172    14347    0.799439775910364    82    2
concoct.72    concoct    36    0    bacteria    2137617    540    4125    0.705882352941177    71    0
maxbin2.109    maxbin    35    1    bacteria    1012324    207    5324    0.659327731092437    69    2
metabat.85    metabat    33    0    bacteria    1947396    60    47599    0.647058823529412    65    0

This file is from the sample output that's included with DAStool (I can also share the output of a run)

Basically, it adds these entries one line at a time in decreasing order of bin_score. The first lines appear fairly quickly, but it starts to go _reallllly_ slowly as the bin_score approaches what I've set for the threshold. It seems to never really get there. We've left it running for like a week before and at the end it went like over a day without adding a new entry, just before reaching the threshold.
When we set a different threshold it just reaches this point of completely slowing down at a different place. We left it running with a threshold of 0.0 for a few days and it went way past the scores it was getting stuck at before but then stalled at like ~0.1

Any idea about why DAS_tool is super slow at the end?

Thanks for your help!

**Comment by cmks** [comment]
This is strange. I have a few theories what might be happening here. Can you please send me (cmksieber@gmail.com) the log file of this run or attach it here so that I can get a better picture?
